metabase/metabase
# Mantine forms and password update form

Changes:
- Creates new `metabase/forms` module (directory)
- Moves `Form`, `FormProvider` and related hooks there without changes
- Adds `FormTextInput`, `FormSubmitButton`, `FormErrorMessage` and others based on `metabase/ui` and `formik`, following the same approach used before for components in `metabase/core`
- Migrates `UserPasswordForm` to use new components

How to verify:
- Gear icon -> Account settings -> Password
- Make sure it's possible to update your password
- All relevant error messages (e.g. weak password, not matching password) should be displayed

<img width="544" alt="Screenshot 2023-09-05 at 20 16 53" src="https://github.com/metabase/metabase/assets/8542534/cfbf6c05-34a8-43cf-9007-8a8f577eebaa">
<img width="550" alt="Screenshot 2023-09-05 at 20 17 02" src="https://github.com/metabase/metabase/assets/8542534/f0bb1c23-cee8-4f6b-8ba1-0096928ed926">
<img width="553" alt="Screenshot 2023-09-05 at 20 17 07" src="https://github.com/metabase/metabase/assets/8542534/2074bced-7bf9-47d5-b43a-a80a534480ca">
